nixos/modules/services/search/kibana.nix

```diff
@@ -182,10 +188,22 @@ in {
     systemd.services.kibana = {
       description = "Kibana Service";
       wantedBy = [ "multi-user.target" ];
+      preStart = let
+        cfgFileRun = "${cfg.dataDir}/config.json";
+      in ''
+        cp -f ${cfgFile} ${cfgFileRun}
+        ${optionalString (cfg.elasticsearch.passwordFile == null) ''
+          sed -e 's,"password":"PASSWORD_REPLACED_PRESTART"\,,,g' -i ${cfgFileRun}
+        ''}
+        ${optionalString (cfg.elasticsearch.passwordFile != null) ''
+          ELASTIC_PASSWORD=$(head -n1 ${cfg.elasticsearch.passwordFile})
+          sed -e "s,PASSWORD_REPLACED_PRESTART,$ELASTIC_PASSWORD,g" -i ${cfgFileRun}
```

Comment:
This fails if there are special characters like `"` in the password. Is there a better approach?

Reply:
I don't think this will fail, bash doesn't interpret the quotes in an expanded string.